[tests-only] check if webdav locks don't influece each other

individual-it opened the pull request:
## Description
add tests to make sure webdav locks don't interact negetivly with each other

## Related Issue
part of https://github.com/owncloud/core/issues/38302


## Motivation and Context
make sure https://github.com/owncloud/enterprise/issues/4355 is not caused by webdav locks

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

individual-it committed:
[tests-only] check if webdav locks don't influece each other

phil-davis commented:
https://drone.owncloud.com/owncloud/core/28320/90/13
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:41
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:42
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:43
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:44
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:79
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:80
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:81
    /drone/src/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToRoot.feature:82
```

2 scenario outlines are failing

individual-it commented:
@phil-davis I've corrected the tests and changed the sharing tests also to not rely anymore on the skeleton folder. The risk of getting confused which folder already exists on the receiver side is too high
